PythonVisualizations/LinkedList.py

```diff
@@ -1,451 +1,585 @@
+        val = self.validateArgument()
+        if val is None:
+            self.setMessage("Input value must be less than 9 characters")
+        elif len(self) >= self.MAX_SIZE:
+            self.setMessage("Error! Linked List is already full.")
+        else:  
+            self.insertElem(val)
+
+
+
+    def clickDelete(self):
+        val = self.validateArgument()
+        msg = None
+        if not self.first:
+            msg = "ERROR: Linked list is empty"
+        elif not val:
+            msg = "Input value must be less than 9 characters" 
+        elif val == -1:
+            result = self.delete()
+            msg = "first node deleted"
+        else:
+            result = self.delete(val)
+            if result != None:
+                msg = "{} Deleted!".format(val)
+            else:
+                msg = "Value {} not found".format(val)
+        self.setMessage(msg)
+        self.clearArgument()
+        
+    def clickNewLinkedList(self):
+        self.canvas.delete('all')
+        self.newLinkedList()
+    
+    def clickGetFirst(self):
+        cur = self.getFirst()
+        if cur!= None: 
+            val = cur.key
+            msg = "The first node is {}".format(val)
+        else: msg = "ERROR: Linked list is empty"
+        self.setMessage(msg)
+        self.clearArgument()
+    
+
+    def makeButtons(self):
+        vcmd = (self.window.register(validate),
+                '%d', '%i', '%P', '%s', '%S', '%v', '%V', '%W')
+        findButton = self.addOperation(
+            "Find", lambda: self.clickFind(), numArguments=1,
+            validationCmd=vcmd)
+        insertButton = self.addOperation(
+            "Insert", lambda: self.clickInsert(), numArguments=1,
+            validationCmd=vcmd)
+        self.addAnimationButtons()
+        deleteButton = self.addOperation(
+            "Delete", lambda: self.clickDelete(), numArguments=0,
+            validationCmd=vcmd, maxRows = 2)
+        newLinkedListButton = self.addOperation(
+            "New", lambda: self.clickNewLinkedList(), 
+            numArguments = 0, validationCmd =vcmd, maxRows = 2)
+        getFirstButton = self.addOperation(
+            "Get First", lambda: self.clickGetFirst(), numArguments = 0,
+            validationCmd=vcmd, maxRows = 2)
+    
+        return [findButton, insertButton, deleteButton, newLinkedListButton, getFirstButton]        
+
+    
+    # validate text entry
+    def validateArgument(self):
+        text = self.getArgument()
+        if text:
+            if text.isdigit():
+                val = int(text)
+                if val< 100000000: return val
+            elif len(text)<9: return text
+        else:
+            return -1
+        
+    def startAnimations(self):
+        self.enableButtons(enable=False)
+        super().startAnimations()    
+        
+    def stopAnimations(self):
+        super().stopAnimations()
+        self.enableButtons(enable=True)
+        self.argumentChanged()           
```

Comment:
These customizations of the methods from VisualizationApp aren't trying to do anything different than the ones in the superclass, but have the adverse effect of leaving the Pause and Stop buttons enabled even when animations are done.  Please remove these definitions.